ironfish/src/rpc/routes/chain/getNetworkHashPower.ts

```diff
@@ -0,0 +1,96 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import * as yup from 'yup'
+import { ValidationError } from '../../adapters'
+import { ApiNamespace, router } from '../router'
+
+export type GetNetworkHashPowerRequest = {
+  lookup?: number // number of blocks to lookup
+  height?: number // estimate network speed at the time of the given height
+}
+
+export type GetNetworkHashPowerResponse = {
+  hashesPerSecond: number
+}
+
+export const GetNetworkHashPowerRequestSchema: yup.ObjectSchema<GetNetworkHashPowerRequest> =
+  yup
+    .object({
+      lookup: yup.number().optional(),
+      height: yup.number().optional(),
+    })
+    .defined()
+
+export const GetNetworkHashPowerResponseSchema: yup.ObjectSchema<GetNetworkHashPowerResponse> =
+  yup
+    .object({
+      hashesPerSecond: yup.number().defined(),
+    })
+    .defined()
+
+router.register<typeof GetNetworkHashPowerRequestSchema, GetNetworkHashPowerResponse>(
+  `${ApiNamespace.chain}/getNetworkHashPower`,
+  GetNetworkHashPowerRequestSchema,
+  async (request, node): Promise<void> => {
+    // default values for lookup and height
+    let lookup = 120
+    let height = -1
+
+    if (request.data?.lookup) {
+      lookup = request.data.lookup
+    }
```

Comment:
nit - In a lot of places in our codebase you'll see null coalescence for things like this:

```
let lookup = request.data?.lookup ?? 120
```

You don't need to make this change, just something you might see around the code!

Reply:
Made the change